### PR TITLE
Opt-remarks for gcc

### PR DESCRIPTION
**Pull request by OfekShilon**
gccoptrem

### CODE REVIEW

File: lib/base-compiler.ts [hunk 1]
```diff
@@ -3135,13 +3136,76 @@ export class BaseCompiler {
...
-        if (this.compiler.demangler) {
-            const result = JSON.stringify(output, null, 4);
+        const mapOptType = (type: string, line: ResultLine): 'Missed' | 'Passed' | 'Analysis' => {
+            if (type === 'missed') return 'Missed';
+            if (type === 'optimized') return 'Passed';
+            if (type === 'note') return 'Analysis';
+
+            // Did we miss any types?
+            SentryCapture(line, `Unexpected opt type: ${type}`);
+            return 'Analysis';
```
Comment (junlarsen) [inline lib/base-compiler.ts:3156]: Nice, good to be notified of unexpected values

File: lib/base-compiler.ts [hunk 2]
```diff
@@ -3135,13 +3136,76 @@ export class BaseCompiler {
 
         return await demangler.process(result);
     }
+    processGccOptInfo(stderr: ResultLine[], compileFileName: string): {remarks: OptRemark[]; newStdErr: ResultLine[]} {
```
Comment (partouf) [inline lib/base-compiler.ts:3139]: missing linefeed

File: lib/base-compiler.ts [hunk 3]
```diff
@@ -3135,13 +3136,76 @@ export class BaseCompiler {
...
+            ({remarks, newStdErr: compilationRes.stderr} = this.processGccOptInfo(
+                compilationRes.stderr,
+                this.compileFilename,
+            ));
+        } else if (compilationRes.optPath) {
+            // clang-like
+            const optRemarksRaw = await fs.readFile(compilationRes.optPath, 'utf8');
+            remarks = processRawOptRemarks(optRemarksRaw, this.compileFilename);
+        }
+        if (remarks.length > 0 && this.compiler.demangler) {
```
Comment (partouf) [inline lib/base-compiler.ts:3207]: LF appreciated

File: lib/base-compiler.ts (hunk 2 above)
Comment (OfekShilon) [reply lib/base-compiler.ts:3139]: right

File: lib/base-compiler.ts (hunk 3 above)
Comment (OfekShilon) [reply lib/base-compiler.ts:3209]: I'm surprised the linters didn't complain

**Comment by mattgodbolt** [comment]
This is rolling out now 🎉 Super cool!

**Pull request merged**